src/Umbraco.Web.BackOffice/Services/IconService.cs

```diff
@@ -104,8 +104,19 @@ private IEnumerable<FileInfo> GetAllIconsFiles()
                 // iterate sub directories of app plugins
                 foreach (var dir in appPlugins.EnumerateDirectories())
                 {
-                    var iconPath = _hostingEnvironment.MapPathContentRoot($"{Constants.SystemDirectories.AppPlugins}/{dir.Name}{Constants.SystemDirectories.AppPluginIcons}");
-                    if (Directory.Exists(iconPath))
+                    // AppPluginIcons path was previoulsy the wrong case, so we first check for the prefered directory 
+                    // and then check the legacy directory. It does mean on windows it'll do a double lookup, but this
```

Comment:
Looks all great but I'm just not sure I understand the double lookup on Windows comment.
If I understand correctly:
- if no icon folder => double check in all cases
- if icon folder => windows will always find it in the first check because it's case insensitive and other OS's might need the second check because of case sensitivity.

Or am I misunderstanding something?

Reply:
No, you are right. It was late an I was tired. 😬😴

Reply:
> if icon folder => windows will always find it in the first check because it's case insensitive and other OS's might need the second check because of case sensitivity.

Windows will likely always find the files, but not guaranteed as they could have toggled on case insensitivity for that path

Reply:
Regardless, this should always work, ultimately there are just a couple of occasions (the folder isn't there, or on case sensitive file systems) that a second check will take place.